azurerm/resource_arm_app_service_certificate.go

```diff
@@ -0,0 +1,260 @@
+	}
+
+	if pfxBlob != "" {
+		decodedPfxBlob, err := base64.StdEncoding.DecodeString(pfxBlob)
+		if err != nil {
+			return fmt.Errorf("Could not decode PFX blob: %+v", err)
+		}
+		certificate.CertificateProperties.PfxBlob = &decodedPfxBlob
+	}
+
+	if keyVaultSecretId != "" {
+		parsedSecretId, err := azure.ParseKeyVaultChildID(keyVaultSecretId)
+		if err != nil {
+			return err
+		}
+
+		keyVaultBaseUrl := parsedSecretId.KeyVaultBaseUrl
+
+		keyVaultId, err := azure.GetKeyVaultIDFromBaseUrl(ctx, vaultClient, keyVaultBaseUrl)
+		if err != nil {
+			return fmt.Errorf("Error retrieving the Resource ID for the Key Vault at URL %q: %s", keyVaultBaseUrl, err)
+		}
+		if keyVaultId == nil {
+			return fmt.Errorf("Unable to determine the Resource ID for the Key Vault at URL %q", keyVaultBaseUrl)
+		}
+
+		certificate.CertificateProperties.KeyVaultID = keyVaultId
+		certificate.CertificateProperties.KeyVaultSecretName = utils.String(parsedSecretId.Name)
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resourceGroup, name, certificate); err != nil {
+		return fmt.Errorf("Error creating/updating App Service Certificate %q (Resource Group %q): %s", name, resourceGroup, err)
+	}
+
+	read, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving App Service Certificate %q (Resource Group %q): %s", name, resourceGroup, err)
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read App Service Certificate %q (Resource Group %q) ID", name, resourceGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmAppServiceCertificateRead(d, meta)
+}
+
+func resourceArmAppServiceCertificateRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).web.CertificatesClient
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resourceGroup := id.ResourceGroup
+	name := id.Path["certificates"]
+
+	ctx := meta.(*ArmClient).StopContext
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] App Service Certificate %q (Resource Group %q) was not found - removing from state", name, resourceGroup)
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on App Service Certificate %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+
+	if props := resp.CertificateProperties; props != nil {
+		d.Set("friendly_name", props.FriendlyName)
+		d.Set("subject_name", props.SubjectName)
+		d.Set("host_names", props.HostNames)
+		d.Set("issuer", props.Issuer)
+		d.Set("issue_date", props.IssueDate.Format(time.RFC3339))
+		d.Set("expiration_date", props.ExpirationDate.Format(time.RFC3339))
+		d.Set("thumbprint", props.Thumbprint)
+		d.Set("key_vault_id", props.KeyVaultID)
```

Comment:
This value doesn't exist in the schema?
```
Test ended in panic.

------- Stdout: -------
=== RUN   TestAccAzureRMAppServiceCertificate_KeyVault
=== PAUSE TestAccAzureRMAppServiceCertificate_KeyVault
=== CONT  TestAccAzureRMAppServiceCertificate_KeyVault

------- Stderr: -------
panic: Invalid address to set: []string{"key_vault_id"}

```

Reply:
Sorry, will run `test` and `acctest` next time to void this type of issues.